paddle/operators/sequence_reshape_op.h

```diff
@@ -0,0 +1,144 @@
+class SequenceReshapeKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& context) const override {
+    auto* in = context.Input<LoDTensor>("X");
+    auto* out = context.Output<LoDTensor>("Out");
+    int out_width = context.Attr<int>("new_dim");
+
+    const T* p_in_data = in->data<T>();
+
+    auto in_dims = in->dims();
+    int64_t in_width = in_dims[1];
+    auto& in_lod = in->lod();
+
+    PADDLE_ENFORCE_EQ(in_lod.size(), 1UL,
+                      "Only support one level sequence now.");
+    PADDLE_ENFORCE_EQ(
+        in_dims[0], in_lod[0].back(),
+        "Inconsistent size between X.shape[0] and X.lod()[0].back().");
+
+    auto in_lod_l0 = in_lod[0];
+    int seq_num = in_lod_l0.size() - 1;
+
+    auto& out_lod = *out->mutable_lod();
+    out_lod.resize(1);
+    out_lod[0].clear();
+    out_lod[0].push_back(0);
+    for (int i = 0; i < seq_num; ++i) {
+      size_t seq_len = in_lod_l0[i + 1] - in_lod_l0[i];
+      size_t offset = 0;
+      offset = (seq_len * in_width) / out_width;
+      PADDLE_ENFORCE_EQ(offset * out_width, seq_len * in_width,
+                        "Please make sure (sequence_length * dimension) can be "
+                        "divided by new_dim with no remainder for each "
+                        "sequence. The %dth sequence is invalid.",
+                        i + 1);
+      PADDLE_ENFORCE_GT(offset, 0,
+                        "Illegal operation, length of the %dth sequence become "
+                        "to 0 after reshaped.",
+                        i + 1);
+      out_lod[0].push_back(out_lod[0].back() + offset);
+    }
+
+    out->mutable_data<T>(context.GetPlace());
+    out->Resize({static_cast<int64_t>(out_lod[0].back()), out_width});
+    T* p_out_data = out->mutable_data<T>(context.GetPlace());
+    math::set_constant(context.device_context(), out, 0.0f);
+
+    for (int i = 0; i < seq_num; ++i) {
+      size_t in_offset = in_lod_l0[i] * in_width;
+      size_t out_offset = out_lod[0][i] * out_width;
+      size_t in_count = (in_lod_l0[i + 1] - in_lod_l0[i]) * in_width;
+      size_t out_count = (out_lod[0][i + 1] - out_lod[0][i]) * out_width;
+      size_t bytes = sizeof(T) * std::min(in_count, out_count);
+      if (platform::is_cpu_place(context.GetPlace())) {
+        memory::Copy(boost::get<platform::CPUPlace>(context.GetPlace()),
+                     p_out_data + out_offset,
+                     boost::get<platform::CPUPlace>(context.GetPlace()),
+                     p_in_data + in_offset, bytes);
+      } else {
+#ifdef PADDLE_WITH_CUDA
+        auto& dev_ctx =
+            context.template device_context<platform::CUDADeviceContext>();
+        memory::Copy(boost::get<platform::CUDAPlace>(context.GetPlace()),
+                     p_out_data + out_offset,
+                     boost::get<platform::CUDAPlace>(context.GetPlace()),
+                     p_in_data + in_offset, bytes, dev_ctx.stream());
+#endif
+      }
+    }
```

Comment:
From the description of the [example](https://github.com/PaddlePaddle/Paddle/pull/7662/files#diff-4eab18b1359b101803e4e19a0bdfc8a0R55), you need only copy input to output and reset `out_lod` and `out_dim`, but not so complex.

Reply:
Thanks